src/openqaoa-qiskit/backends/qaoa_qiskit_qpu.py

```diff
@@ -164,7 +182,14 @@ def parametric_qaoa_circuit(self) -> QuantumCircuit:
 
         if self.append_state:
             parametric_circuit = parametric_circuit.compose(self.append_state)
-        parametric_circuit.measure_all()
+        
+        #only measure the problem qubits
+        if self.final_mapping is None:
+            parametric_circuit.measure(self.problem_reg,creg)
+        else:
+            for idx,qubit in enumerate(self.final_mapping[0:len(self.problem_reg)]):
+                    cbit = creg[idx]
+                    parametric_circuit.measure(qubit,cbit)
```

Comment:
wrong indentation I think*

Reply:
I don't know if Github is rendering this weirdly -- the indentation seems to be fine on my editor.